inference.py

```diff
@@ -10,169 +10,127 @@
+                    augmented_input = transformer.augment_image(inputs)
+                    augmented_output = model(augmented_input)
+                    if isinstance(augmented_output, OrderedDict) and 'out' in augmented_output.keys():
+                        augmented_output = augmented_output['out']
+                    # reverse augmentation for outputs
+                    deaugmented_output = transformer.deaugment_mask(augmented_output)
+                    deaugmented_output = F.softmax(deaugmented_output, dim=1).squeeze(dim=0)
+                    output_lst.append(deaugmented_output)
+                outputs = torch.stack(output_lst)
+                outputs = torch.mul(outputs, WINDOW_SPLINE_2D)
+                outputs, _ = torch.max(outputs, dim=0)
+                outputs = outputs.permute(1, 2, 0).argmax(dim=-1)
+                outputs = outputs.reshape(sample_size, sample_size).cpu().numpy()
+                pred_img[row:row + sample_size, col:col + sample_size] = outputs
+    pred_img = pred_img[padding:-padding, padding:-padding]
+    gdf = None
+    if label_arr is not None:
+        feature = defaultdict(list)
+        cnt = 0
+        for row in tqdm(range(0, h, sample_size), position=1, leave=False, desc='Inferring rows'):
+            with tqdm(range(0, w, sample_size), position=2, leave=False, desc='Inferring columns') as _tqdm:
+                for col in _tqdm:
+                    label = label_arr[row:row + sample_size, col:col + sample_size]
+                    pred = pred_img[row:row + sample_size, col:col + sample_size]
+                    pixelMetrics = ComputePixelMetrics(label.flatten(), pred.flatten(), num_classes)
+                    eval = pixelMetrics.update(pixelMetrics.iou)
+                    feature['id_image'].append(gpkg_name)
+                    for c_num in range(num_classes):
+                        feature['L_count_' + str(c_num)].append(int(np.count_nonzero(label == c_num)))
+                        feature['P_count_' + str(c_num)].append(int(np.count_nonzero(pred == c_num)))
+                        feature['IoU_' + str(c_num)].append(eval['iou_' + str(c_num)])
+                    feature['mIoU'].append(eval['macro_avg_iou'])
+                    x_1, y_1 = (xmin + (col * xres)), (ymax - (row * yres))
+                    x_2, y_2 = (xmin + ((col * xres) + mx)), y_1
+                    x_3, y_3 = x_2, (ymax - ((row * yres) + my))
+                    x_4, y_4 = x_1, y_3
+                    geom = Polygon([(x_1, y_1), (x_2, y_2), (x_3, y_3), (x_4, y_4)])
+                    feature['geometry'].append(geom)
+                    feature['length'].append(geom.length)
+                    feature['pointx'].append(geom.centroid.x)
+                    feature['pointy'].append(geom.centroid.y)
+                    feature['area'].append(geom.area)
+                    cnt += 1
+        gdf = gpd.GeoDataFrame(feature, crs=input_image.crs)
+        gdf.to_crs(crs="EPSG:4326", inplace=True)
```

Comment:
could/should this be extracted as a standalone function?

Reply:
opened to syntax corrections!